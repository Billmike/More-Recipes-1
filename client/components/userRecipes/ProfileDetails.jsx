import React from 'react';
import PropTypes from 'prop-types';
import ReactTooltip from 'react-tooltip';

const propTypes = {
  user: PropTypes.shape({
    userProfile: PropTypes.shape({
      username: PropTypes.string,
      firstName: PropTypes.string,
      lastName: PropTypes.string,
      email: PropTypes.string,
      aboutMe: PropTypes.string,
    })
  }).isRequired,
  handleStartEdit: PropTypes.func.isRequired,
  userId: PropTypes.number.isRequired,
  authenticatedUserId: PropTypes.number
};

const defaultProps = {
  authenticatedUserId: null,
};

const ProfileDetails = (props) => {
  const { user: { userProfile }, handleStartEdit, userId,
    authenticatedUserId } = props;
  const { username, firstName, lastName, email, aboutMe } = userProfile;
  const editableDetails = {
    firstName,
    lastName,
    aboutMe,
  };

  const handleEdit = () => {
    handleStartEdit(editableDetails);
  };

  return (
    <div>
      <div className="row" />
      <div className="col s12">
        <h5><i className="material-icons left">person</i>
          {userId === authenticatedUserId ? 'My' : firstName} Profile</h5>
        <div className="divider black" />
        <br />
        <div id="about_user" className="row">
          <div id="profile_username" className="row">
            <label htmlFor="username" className="col s3 black-text">
              Username:
            </label>
            <div id="username" className="col s9">{username}</div>
          </div>
          <div id="profile_first_name" className="row">
            <label htmlFor="firstname" className="col s3 black-text">
              First Name:
            </label>
            <div id="firstname" className="col s9">{firstName}</div>
          </div>
          <div id="profile_last_name" className="row">
            <label htmlFor="lastname" className="col s3 black-text">
              Last Name:
            </label>
            <div id="lastname" className="col s9">{lastName}</div>
          </div>
          <div id="profile_email" className="row">
            <label htmlFor="email" className="col s3 black-text">
              Email:
            </label>
            <div id="email" className="col s9">{email}</div>
          </div>
        </div>
      </div>
      <div className="col s12">
        <h5>About {userId === authenticatedUserId ? 'Me' : firstName}</h5>
        <div className="divider black" />
        <p
          id="about-me"
        >
          {aboutMe}
        </p>
      </div>
      <div>
        {userId === authenticatedUserId
        && <button
          id="edit-btn"
          className={'btn-floating waves-effect waves-light green right'}
          onClick={handleEdit}
        >
          <i className="material-icons left" data-tip="Edit Details">edit</i>
        </button>}
        <ReactTooltip />
      </div>
      <div className="row" />
    </div>
  );
};

ProfileDetails.propTypes = propTypes;

ProfileDetails.defaultProps = defaultProps;

export default ProfileDetails;
