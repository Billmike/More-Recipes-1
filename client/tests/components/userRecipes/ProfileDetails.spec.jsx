import React from 'react';

import ProfileDetails from '../../../components/userRecipes/ProfileDetails';

describe('ProfileDetails component', () => {
  it('should render component', () => {
    const props = {
      user: {
        userProfile: {
          username: 'johnmark',
          firstName: 'John',
          lastName: 'Mark',
          email: 'john@example.com',
          aboutMe: 'This is about me',
        },
      },
      handleStartEdit: jest.fn(),
      currentProfileUserId: 1,
      authenticatedUserId: null,
    };
    const wrapper = shallow(<ProfileDetails {...props} />);

    expect(wrapper).toBeDefined();
  });
  it('should simulate click action', () => {
    const props = {
      user: {
        userProfile: {
          username: 'johnmark',
          firstName: 'John',
          lastName: 'Mark',
          email: 'john@example.com',
          aboutMe: 'This is about me',
        },
      },
      handleStartEdit: jest.fn(),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
    };
    const wrapper = shallow(<ProfileDetails {...props} />);

    expect(wrapper).toBeDefined();
    wrapper.find('button#edit-btn').simulate('click');
    expect(props.handleStartEdit).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
