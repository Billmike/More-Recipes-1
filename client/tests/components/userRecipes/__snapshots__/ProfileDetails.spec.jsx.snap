// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDetails component should simulate click action 1`] = `
<div>
  <div
    className="row"
  />
  <div
    className="col s12"
  >
    <h5>
      <i
        className="material-icons left"
      >
        person
      </i>
      My
       Profile
    </h5>
    <div
      className="divider black"
    />
    <br />
    <div
      className="row"
      id="about_user"
    >
      <div
        className="row"
        id="profile_username"
      >
        <label
          className="col s3 black-text"
          htmlFor="username"
        >
          Username:
        </label>
        <div
          className="col s9"
          id="username"
        >
          johnmark
        </div>
      </div>
      <div
        className="row"
        id="profile_first_name"
      >
        <label
          className="col s3 black-text"
          htmlFor="firstname"
        >
          First Name:
        </label>
        <div
          className="col s9"
          id="firstname"
        >
          John
        </div>
      </div>
      <div
        className="row"
        id="profile_last_name"
      >
        <label
          className="col s3 black-text"
          htmlFor="lastname"
        >
          Last Name:
        </label>
        <div
          className="col s9"
          id="lastname"
        >
          Mark
        </div>
      </div>
      <div
        className="row"
        id="profile_email"
      >
        <label
          className="col s3 black-text"
          htmlFor="email"
        >
          Email:
        </label>
        <div
          className="col s9"
          id="email"
        >
          john@example.com
        </div>
      </div>
    </div>
  </div>
  <div
    className="col s12"
  >
    <h5>
      About 
      Me
    </h5>
    <div
      className="divider black"
    />
    <p
      id="about-me"
    >
      This is about me
    </p>
  </div>
  <div>
    <button
      className="btn-floating waves-effect waves-light green right"
      id="edit-btn"
      onClick={[Function]}
    >
      <i
        className="material-icons left"
        data-tip="Edit Details"
      >
        edit
      </i>
    </button>
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    />
  </div>
  <div
    className="row"
  />
</div>
`;
